crates/predictor/src/predictor/point.rs

```diff
@@ -103,59 +109,118 @@ impl Point {
-                best_level_diff = diff;
-            }
-        }
-
-        // Round to directional DATA_RESOLUTION
-        let mangled_lat = self.latitude / DATA_RESOLUTION;
-        let mangled_lon = self.longitude / DATA_RESOLUTION;
-
-        let percent_north = mangled_lat.ceil() - mangled_lat.floor();
-        let percent_east = mangled_lon.ceil() - mangled_lon.floor();
-
-        Alignment {
-            ne: AlignedPoint {
-                latitude: Point::align_lat(mangled_lat.ceil()),
-                longitude: Point::align_lon(mangled_lon.ceil()),
-                level: best_level
-            },
-            nw: AlignedPoint {
-                latitude: Point::align_lat(mangled_lat.ceil()),
-                longitude: Point::align_lon(mangled_lon.floor()),
-                level: best_level
-            },
-            se: AlignedPoint {
-                latitude: Point::align_lat(mangled_lat.floor()),
-                longitude: Point::align_lon(mangled_lon.ceil()),
-                level: best_level
-            },
-            sw: AlignedPoint {
-                latitude: Point::align_lat(mangled_lat.floor()),
-                longitude: Point::align_lon(mangled_lon.floor()),
-                level: best_level
-            },
-            percent_north: percent_north,
-            percent_south: 1.0 - percent_north,
-            percent_east: percent_east,
-            percent_west: 1.0 - percent_east
-        }
-
-    }
+     pub fn align(&self) -> Alignment {
+         let isobaric_hpa = 1013.25*(1.0 - self.altitude/44330.0).powf(5.255);
+
+         //TODO: make a fast lookup structure for this
+         let levels = [2, 3, 5, 7, 10, 20, 30, 50, 70, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700, 750, 800, 850, 900, 925, 950, 975, 1000];
+         let mut best_level : i32 = 1;
+         let mut best_level_diff : f32 = (isobaric_hpa - (best_level as f32)).abs();
+         let mut best_level_index : usize = 0;
+         let mut curr_index : usize = 0;
+
+         for level_ref in levels.iter() {
+             let level = *level_ref as i32;
+             let diff = (isobaric_hpa - (level as f32)).abs();
+
+             if diff < best_level_diff {
+                 best_level = level;
+                 best_level_diff = diff;
+                 best_level_index = curr_index;
+             }
+             curr_index = curr_index + 1;
+         }
+
+         // Determine which level is up and which is down
+         let mut level_up : i32 = 0;
+         let mut level_up_diff : f32 = 0.0;
+         let mut level_down : i32 = 0;
+         let mut level_down_diff : f32 = 0.0;
+
+         let mut alt_floor :f32 = 0.0;
+         let mut alt_ceil : f32 = 1000.0;
+
+         if isobaric_hpa < alt_floor || isobaric_hpa > alt_ceil {
+             level_up = best_level;
+             level_down = best_level;
+             level_up_diff = best_level_diff;
+             level_down_diff = best_level_diff;
```

Comment:
I did this in the event that isobaric_hpa > alt_ceil. best_level would be 1000 hpa and not 0. So have to update them both to be max values rather than 0.